pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -179,6 +184,8 @@ type ConfigGlobal struct {
 	// located in (i.e. where the controller will be running). If this is
 	// blank, then the local zone will be discovered via the metadata server.
 	LocalZone string `gcfg:"local-zone"`
+	// InstanceName specifies the GCE instance name of the node
+	InstanceName string `gcfg:"instance-name"`
```

Comment:
is this the source of truth for the instance name? expected to be consumed by the kubelet and registered in the Node object for use by the cloud controller? is there any way for the controller to verify from information in the instance that Node X really does correspond to it?

Reply:
Yes, this is the source of truth for instance name and is expected to be consumed by kubelet. I cannot think of any straightforward way to verify the information that node X really does correspond to it. Which is the reason why I was thinking that I should close this PR.

It does annoy me that node name cannot be different from IaaS identifiers as of now. But until we can think of anything else I guess operators will have to live with it.